### DISCUSSION TITLE
Holiday Calendar event in Cell Header?

### DISCUSSION THREAD

**Discussion by memoryone85** [MMRIZE/MMM-CalendarExt3#178] [Q&A]
Hello;
Is it possible to put a holiday calendar inline with the date header cell instead as a seperate event?
Thanks.

**Comment by eouia** [comment]
The basic concept
https://github.com/MMRIZE/MMM-CalendarExt3/wiki/Set-cell-background-instead-of-holiday-event

To apply to your case;
1) name the calendar in the `default calendar` module.
```js
{
	symbol: "calendar-check",
	url: "https://ics.calendarlabs.com/76/mm3137/US_Holidays.ics",
	color: "red",
	name: "Holiday"
},
```

2) Add these code into your CX3 config.
```js

eventTransformer: (ev) => {
	if (ev.calendarName === 'Holiday') ev.skip = true
	return ev
},
manipulateDateCell: (cellDom, events) => {
	if (!Array.isArray(events)) return
	const holiday = events.find(e => e.calendarName === 'Holiday')
	if (holiday) {
		const cellHeader = cellDom.querySelector('.cellHeader') ?? null
		if (cellHeader) cellHeader.style.background = holiday.color
	}
},
```

**Before**
<img width="450" alt="image" src="https://github.com/user-attachments/assets/19de2d7d-3f7f-4b23-85ac-cc7f185176bd">

**After**
<img width="450" alt="image" src="https://github.com/user-attachments/assets/0f7f46ae-fc29-4b51-ab06-6eac47b2c550">

**Comment by memoryone85** [comment]
Awesome, This worked perfectly! Thanks so much!

EDIT: hm I didnt realize that it just colors the day. any chance we can also get the title of the holiday too?

**Reply by eouia** [reply]
I missed your edited question. 

```js
manipulateDateCell: (cellDom, events) => {
	if (!Array.isArray(events)) return
	const holiday = events.find(e => e.calendarName === 'Holiday')
	if (holiday) {
		const cellHeader = cellDom.querySelector('.cellHeader') ?? null
		if (cellHeader) {
			cellHeader.style.background = holiday.color
			const cellTitle = document.createElement('div')
			cellTitle.classList.add('cellTitle')
			cellTitle.textContent = holiday.title
			cellHeader.insertBefore(cellTitle, cellHeader.childNodes[1])
		}
	}
},
```

<img width="336" alt="image" src="https://github.com/user-attachments/assets/524d5c10-147d-4b97-923e-01398251e15e" />


This is not a complete solution (Because there could be some exceptional cases, such as an event title that is too long or multiple holidays on the same date).
